content/articles/functional-programming-with-java-lambda/index.md

```diff
@@ -0,0 +1,333 @@
+
+Lambda Stream pipeline is a sequence of processing steps from the data source (from collections). The pipeline performs two main operations which is an intermediate or terminal operation. Which formed a chain method calls.
+
+#### Intermediate and terminal Operation
+An intermediate operation is an activity that plays out a particular task on a stream element and the result of the task form a new stream. Intermediate operations are also called Lazy operations. Which implies that the operation isn't performed until a terminal operation is called.
+
+| Operation  |                           Description                           |
+| ---------- | :-------------------------------------------------------------: |
+| filter     | The filter method allows us to get elements of the stream that  |
+|            |                     meet certain conditions.                     |
+| ---------- | :-------------------------------------------------------------: |
+| distinct   |   The distinct method returns distinct elements of the stream.    |
+| ---------- | :-------------------------------------------------------------: |
+| limit      |  The limit method return elements of the stream with the given  |
+|            |                         number or limit.                        |
+| ---------- | :-------------------------------------------------------------: |
+| map        |  The filter method allows us to get elements of the stream that  |
+|            |                     meet certain conditions.                     |
+| ---------- | :-------------------------------------------------------------: |
+| sorted     |         Returns elements of the stream in a given order.          |
+| ---------- | :-------------------------------------------------------------: |
+
+#### The intermediate operation
+A Terminal Operation starts processing of intermediate operation performed on stream and produces the result. Terminal operations are known as eager operations because they carry out the requested task when they are called.
+
+| Operation  |                           Description                           |
+| ---------- | :-------------------------------------------------------------: |
+| forEach    |       Loops through the stream to operate on each element.        |
+| ---------- | :-------------------------------------------------------------: |
+| count      |     Returns the total number of elements that operations are      |
+|            |                          performed on.                           |
+| ---------- | :-------------------------------------------------------------: |
+| average    | Returns the average of element numerical elements of the stream.  |
+|            |                                                                 |
+| ---------- | :-------------------------------------------------------------: |
+| min        |       Returns the minimum element of numbers in the stream.      |
```

Comment:
Change `Returns the minimum element of numbers in the stream`  to `Returns the minimum number of elements in the stream`. Do the same for the others.